controller/controlplane/controller.go

```diff
@@ -340,6 +348,28 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		log.Debug(logger, "DataPlane not set, deployment for ControlPlane will remain dormant")
 	}
 
+	log.Trace(logger, "validating ReferenceGrants exist for the ControlPlane")
+	if err := r.validateReferenceGrants(ctx, cp); err != nil {
+		k8sutils.SetCondition(
+			k8sutils.NewCondition(
+				kcfgdataplane.ReadyType,
+				metav1.ConditionFalse,
+				kcfgcontrolplane.ConditionReasonMissingReferenceGrant,
+				"ReferenceGrant(s) are missing for the ControlPlane",
+			),
+			cp,
+		)
+		res, err := r.patchStatus(ctx, logger, cp)
+		if err != nil {
+			log.Debug(logger, "unable to patch ControlPlane status", "error", err)
+			return ctrl.Result{}, err
+		}
+		if !res.IsZero() {
+			return res, nil
+		}
+		return ctrl.Result{}, nil
```

Comment:
I think this could be simplified to the below snippet - the error will be logged anyway if we return an error from the `Reconcile`.
```suggestion
		if err != nil || !res.IsZero() {
			return res, err
		}
		return ctrl.Result{}, nil
```